src/_P001_Switch.ino

```diff
@@ -273,7 +273,12 @@ boolean Plugin_001(byte function, struct EventStruct *event, String& string)
           //setPinState(PLUGIN_ID_001, CONFIG_PIN1, PIN_MODE_INPUT, switchstate[event->TaskIndex]);
           //  if it is in the device list we assume it's an input pin
           if (Settings.TaskDevicePin1PullUp[event->TaskIndex]) {
-            pinMode(CONFIG_PIN1, INPUT_PULLUP);
+            #if defined(ESP8266)
+            if (CONFIG_PIN1 == 16)
+              pinMode(CONFIG_PIN1, INPUT_PULLDOWN_16);
+            else
+            #endif
+              pinMode(CONFIG_PIN1, INPUT_PULLUP);
```

Comment:
I had to read this twice to see the `else` was in the `#ifdef ... #endif` scope.
It is more clear to assign the pin mode to some variable and then in the  `#ifdef ... #endif` scope check for the pin number and set another mode.
Then outside the  `#ifdef ... #endif` scope you can set the `pinMode`.

Reply:
done